SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -256,7 +256,7 @@ void USpatialActorChannel::RetireEntityIfAuthoritative()
 			}
 			else
 			{
-				Sender->RetireEntity(EntityId, Actor->IsNetStartupActor());
+				NetDriver->ActorSystem->RetireEntity(EntityId, Actor->IsNetStartupActor());
```

Comment:
Changes such as these seem rather unfortunate to me as ActorSystem (in my impression at least) has a single responsibility of mapping Spatial's entities conveniently to UE's actors, and isn't the one responsible for retiring entities, or updating interest (below in this file). Do we have a task to separate this, too?

Reply:
Great question, and I agree not all of these fit the Actor system's current responsibilities. I don't think these are too far off that this is egregious to the point that we shouldn't do it- i.e. both retiring entities and updating Interest involve interacting with entities in order to map Actors to entities- but this could be better. I think making this cleaner falls under https://improbableio.atlassian.net/browse/UNR-4851 (especially if that will be split into more tasks as it looks like it will currently).

Reply:
Agree 💯 , thanks for the clarification.